File: soyeong/programmers-2/N개의-최소공배수.py
def solution(arr):
    # 배열 수 중 보다 가장 큰 수를 기준으로 배수를 구하고
    # 커지는 배수마다 배열의 수로 나누어 모두 나누어 떨어지는 순간의 수를 최소공배수

    max_num = max(arr)  # 가장 큰 수
    i = 1
    while True:
        multi_num = max_num * i  # 큰 수의 배수
        flag = True  # 배수 판별 플래그
        for one in arr:
            if multi_num % one != 0:  # 하나의 수라도 나누어 떨어지지 않으면 false
                flag = False

        if flag:  # 모두 나누어 떨어지면 true
            return multi_num  # 최소공배수 return
        else:
            i += 1

File: ingwang/programmers-2/N개의 최소공배수.java
class Solution18 {
    public static int solution18(int[] arr) {
        int answer = 0;

        if (arr.length == 1) {
            return arr[0];
        }
        // 최대공약수
        int m = max(arr[0], arr[1]);
        // 최대 공약수
        int n = (arr[0] *arr[1])/m;

        if(arr.length>2){
            for (int i = 2; i < arr.length; i++) {
                m = max(n, arr[i]);
                n = (n * arr[i])/m;
            }
        }
        answer = n;
        return answer;
    }

    private static int max(int n, int m) {
        int x = n % m;
        if (x == 0) {
            return m;
        }
        return max(m, x);
        
    }

    public static void main(String[] args) {
        int [] a = {1,2,3};
        System.out.println(solution18(a));
    }
}
